main.go

```diff
@@ -138,26 +180,63 @@ func main() {
 
 	ctx := ctrl.SetupSignalHandler()
 
-	tlsOptions, err := common.GetSspTlsOptions(ctx)
+	tlsCfgFunc := func(cfg *tls.Config) {
```

Comment:
Is it possible to keep that function outside of `main()`? I think it's a good practice to keep `main()` short function, it's easier to read the code that way.